src/components/SeatMap/api.test.js

```diff
@@ -0,0 +1,196 @@
+    })
+
+    it('should call postData with the correct parameters', async () => {
+      const flightFixture = flightDetails();
+
+      await api.getPlaneFeatures(flightFixture, 'EN', 'metric');
+
+      expect(api.postData).toHaveBeenCalledWith(
+        'flight/features/plane/seatmap',
+        { flight: flightFixture, lang: 'EN', units: 'metric' },
+      );
+    });
+
+    it('should call postData with the default language if the provided language is not supported', async () => {
+      const flightFixture = flightDetails();
+
+      await api.getPlaneFeatures(flightFixture, 'A', 'metric');
+
+      expect(api.postData).toHaveBeenCalledWith(
+        'flight/features/plane/seatmap',
+        { flight: flightFixture, lang: 'EN', units: 'metric' },
+      );
+    })
+  });
+
+  describe('for a specific cabin class (F, B, P or E)', () => {
+    it('should return the correct information when all the information is present', async () => {
+      const flightFixture = flightDetails({ cabinClass: 'E' });
+      const singleCabinResponseFixture = [{
+        id: '1111',
+        cabin: cabin(),
+        entertainment: entertainment(),
+        power: power(),
+        wifi: wifi(),
+        seatDetails: seatDetails(),
+      }];
+      api.postData.mockImplementation(() => singleCabinResponseFixture);
+
+      const result = await api.getPlaneFeatures(flightFixture);
+
+      expect(result).toEqual({
+        E: {
+          cabin: singleCabinResponseFixture[0].cabin,
+          entertainment: singleCabinResponseFixture[0].entertainment,
+          power: singleCabinResponseFixture[0].power,
+          wifi: singleCabinResponseFixture[0].wifi,
+        },
+        seatDetails: singleCabinResponseFixture[0].seatDetails,
+      });
+    });
+
+    it('should throw if no seat details were returned from the API', async () => {
+      const flightFixture = flightDetails({ cabinClass: 'E' });
+      const singleCabinResponseFixture = [{
+        id: '1111',
+        cabin: cabin(),
+        entertainment: entertainment(),
+        power: power(),
+        wifi: wifi(),
+      }];
+      api.postData.mockImplementation(() => singleCabinResponseFixture);
+
+      await expect(api.getPlaneFeatures(flightFixture)).rejects.toThrow('data is not found for the flight: 1111');
+    });
+
+    it('should throw if the response item has an error', async () => {
+      const flightFixture = flightDetails({ cabinClass: 'E' });
+      const singleCabinResponseFixture = [{
+        id: '1111',
+        cabin: cabin(),
+        entertainment: entertainment(),
+        power: power(),
+        wifi: wifi(),
+        error: 'Something went wrong!',
+      }];
+      api.postData.mockImplementation(() => singleCabinResponseFixture);
+
+      await expect(api.getPlaneFeatures(flightFixture)).rejects.toThrow('Something went wrong!');
+    });
+  });
+
+  describe('for the whole plane (cabin class A is sent in request)', () => {
```

Comment:
Can we say in this case 'when the whole plane.....' this is to be consistent with the when/should pattern which is what we usually add in our new microfrontends